cnf-tests/testsuites/pkg/networks/network.go

```diff
@@ -31,42 +54,50 @@ func NewNetworkAttachmentDefinitionBuilder(namespace, nadName string) *NetworkAt
 }
 
 func (b *NetworkAttachmentDefinitionBuilder) WithAdditonalPlugin(config string) *NetworkAttachmentDefinitionBuilder {
-	b.configs = append(b.configs, config)
+	b.appendConfig(config)
 	return b
 }
 
 func (b *NetworkAttachmentDefinitionBuilder) WithTuning(sysctls string) *NetworkAttachmentDefinitionBuilder {
-	b.configs = append(b.configs, sysctls)
+	b.appendConfig(sysctls)
 	return b
 }
 
 func (b *NetworkAttachmentDefinitionBuilder) WithBond(bondName, link1, link2 string) *NetworkAttachmentDefinitionBuilder {
-	b.configs = append(b.configs, fmt.Sprintf(`{
-	    "type": "bond",
-		"ifname": "%s",
-		"mode": "active-backup",
-		"failOverMac": 1,
-		"linksInContainer": true,
-		"miimon": "100",
-		"links": [ {"name": "%s"}, {"name": "%s"} ],
-        "ipam": {
-            "type": "host-local",
-            "subnet": "1.1.1.0/24"
-        }}`, bondName, link1, link2))
+	b.appendConfig(fmt.Sprintf(bondConfig, bondName, link1, link2))
+	return b
+}
+
+func (b *NetworkAttachmentDefinitionBuilder) WithStaticIpam(ip string) *NetworkAttachmentDefinitionBuilder {
+	b.tokens[len(b.tokens)-1].ipam = fmt.Sprintf(staticIpam, ip)
```

Comment:
Here we assume tokens has at least one item, which is not always true

Reply:
fixed